pkg/logql/ast.go

```diff
@@ -998,6 +998,10 @@ func (e *BinOpExpr) String() string {
 
 // impl SampleExpr
 func (e *BinOpExpr) Shardable() bool {
+	if e.opts != nil && e.opts.VectorMatching != nil {
```

Comment:
@owen-d 
We cannot use `*e.opts.VectorMatching != zero` to check whether VectorMatching struct is empty, it is because `VectorMatching` has `Include []string`, it will cause `Invalid operation: *e.opts.VectorMatching != zero (the operator != is not defined on VectorMatching)`.
I thought `e.opts != nil && e.opts.VectorMatching != nil` should be good enough? Any thoughts, please advise! Thanks!
```
type BinOpOptions struct {
	ReturnBool     bool
	VectorMatching *VectorMatching
}
```

Reply:
added sharding test - https://github.com/grafana/loki/pull/4391/commits/0ab8264a4fa47e0ced8ddc3e1e07ee1b3ad338fe and https://github.com/garrettlish/loki/blob/onOrIgnoring/pkg/logql/shardmapper_test.go#L984, could you please help review it again? Thanks!

Reply:
Ah bummer, slice comparison :sob: . Anyway, `e.opts != nil && e.opts.VectorMatching != nil` should be fine. LGTM!